### ISSUE TITLE
tue-get update bug

### ISSUE THREAD

**Issue by ar13pit** [tue-robotics/tue-env#487]
**Bug Description**
If a new function is added to either tue-functions.bash (and exported) or tue-install.bash, it isn't available to tue-install-impl.bash for use during the update process. 

**Proposal**
1. First update tue-env and then start a new bash shell within tue-install-impl.bash to have all the updates available.
2. Split update into 2 steps, one where only tue-env gets update and second where only packages and targets repos get updated.
3. Do not add any function on which update process is dependent in tue-functions.bash or tue-install.bash.

**Comment by MatthijsBurgh** [comment]
@ar13pit We have introduced this bug ourselves. As we added functions outside `tue-install-impl`, which are used inside `tue-install-impl.`
So starting a new bash shell is too much IMO. As we move shared functions to a separate file. This file is sourced in the normal setup, But also again in `tue-install-impl`. Then you are sure, than the up-to-date versions of the functions are available.

I don't think point 2 is desired. I think the current setup is good enough, but we screwed it up.

**Comment by ar13pit** [comment]
True, we did introduce this but it was bound to happen sooner or later. Point 1 is easy to implement actually, in a nutshell by doing something like `bash -c CMD` from within the script. But lets not do that now as we have nothing broken at the moment.

The reason why I will leave this issue open is I want to check if this holds even upon migration to python.
